### DISCUSSION TITLE
can I get limited number of included results in prisma?

### DISCUSSION THREAD

**Discussion by Knorway** [prisma/prisma#9321] [Q&A] [answered]
```
pseudo prisma code

const user = await prisma.user.findMany({
	where: {
		id: req.params.id,
	},
	include: {
		chatRooms: {
			include: {
				messges: true,
				limit: 5,
			},
		},
	},
});
```
Hi prisma people! I haven't used prisma yet but want to check if it's achievable with prisma so my question is like this. there is a user data that has all the chat rooms it’s participating in and I want to get just limited number of messages from each room. I failed this one in typeorm and am ready to immigrate. thank you for any advice in advance!

**Comment by nahtnam** [comment, marked as answer]
Hello,

Let me try to help here. I believe it is possible to do this with Prisma. I was able to generate the following query:

```
return prisma.User.findMany({
  where: {
    id: '123',
  },
  include: {
    ChatRoom: {
      include: {
        messages: {
          take: 5,
        },
      },
    },
  },
});
```

Based on what you said in your text, ^ is the query you want. However, in the code you provided it seems like you might have wanted to show only 5 rooms with all the messages? That is also possible as well, just let me know and I can help with that.

On a side note, it might be a good idea to sort the messages by `createdAt` so that you get the 5 latest messages as part of this query

**Reply by Knorway** [reply]
thank you so much this is exactly what I was looking for!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
